thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/util/CacheUtils.java

```diff
@@ -86,4 +93,30 @@ public static String buildQuery(JsonObject parameters) {
         parameters.getLong("start"),
         parameters.getLong("end"));
   }
+
+  /**
+   * Convert list of strings to their proper URL hosts. For each
+   * string, it will parse out the proper host for the URI.
+   * Example: "http://localhost:8091" -> "localhost"
+   * @param bootstrapUris
+   * @return
+   */
+  public static List<String> getBootstrapHosts(List<String> bootstrapUris) {
+    List<String> bootstrapHosts = new ArrayList<>(bootstrapUris.size());
+    for (String bootstrapUri : bootstrapUris) {
+      try {
+        URI uri = new URI(bootstrapUri);
+        // The next workaround is to correctly parse urls that don't include any schema
+        if (uri.getHost() == null) {
+          uri = new URI("http://" + bootstrapUri);
+        }
+        String host = uri.getHost();
```

Comment:
Is it possible the host is still null?

Reply:
Good catch. Yes, but Couchbase will throw an exception upon trying to connect to "null". I will add a check. Do you think it would be better to stop execution if there was an invalid host, or just warn the user and skip that particular host?

Reply:
warn and skip

Reply:
done